### PR TITLE
NEXUS-22136 REST API for Helm repositories

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-22136

### CODE REVIEW

File: nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  protected Response put(final String url, final Object body) throws Exception {
+    HttpPut request = new HttpPut();
+    prepareRequest(request, url, body);
+    return execute(request);
+  }
+
+  private Response execute(HttpEntityEnclosingRequestBase request) throws Exception {
+    try (CloseableHttpClient client = super.clientBuilder().build()) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: you can use one try block

File: nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  protected String getCreateRepositoryPathFor(final String type) {
+    return new StringJoiner("/")
+        .add(REPOSITORIES_API_URL)
+        .add(FORMAT_VALUE)
+        .add(type)
+        .toString();
+  }
+
+  protected String getUpdateRepositoryPathFor(final String type, final String name) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: getUpdateRepositoryPathForUrl ?

File: nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    Role role = securityRule.createRole(randomRoleName, UNRELATED_PRIVILEGE);
+    securityRule.createUser(randomRoleName, randomRoleName, role.getRoleId());
+    credentials = new UsernamePasswordCredentials(randomRoleName, randomRoleName);
+  }
+
+  protected void setBadCredentials() {
+    credentials = new UsernamePasswordCredentials("fake_user", "fake_user");
+  }
+
+  protected String getCreateRepositoryPathFor(final String type) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: getCreateRepositoryPathUrl

Comment (mlukaretkyi) [reply nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: fixed

File: nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java (hunk 2 above)
Comment (mlukaretkyi) [reply nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: fixed

File: nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java (hunk 1 above)
Comment (mlukaretkyi) [reply nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/api/ResourceITSupport.java]: fixed

**Pull request merged**